tutorials/e-commerce-store-codelab/step-07-checking-out.md

```diff
@@ -0,0 +1,167 @@
+---
+description: Implementing checking out and order placement for an e-commerce app built with Remix, Drizzle, Turso, and Cloudflare Workers
+keywords:
+  - turso
+  - remix
+  - e-commerce
+---
+
+# Step 7: Checking out and placing orders
+
+After users have selected items they want to buy and placed them in the cart, we
+then expect them to checkout. To facilitate this inside the app, let’s create
+the checkout page by adding a `checkout._index.tsx` file under `app/routes`.
+
+For the loader, we are trying to get the same data as we did in the cart page, only that here we need the user information too. The user data is used to pre-fill some fields on the checkout form.
+
+```tsx
+cartItems = await db.query.cartItems.findMany({
+  where: (cartItems, { eq }) => eq(cartItems.userId, userId),
+  columns: { count: true, id: true },
+  with: { product: true, user: true },
+});
+return {
+  cartItems,
+  userData: cartItems[0].user,
+};
+```
+
+We then get the loader data and lay out the cart items and checkout information
+on the page component. [Click here] to see the page markup of the checkout page
+component.
```

Comment:
Avoid link text saying "click here".  Instead, link the text that describes what the destination is.